File: Backend/PackageService/src/main/java/com/gl/app/PackageService/service/PackageService.java
package com.gl.app.PackageService.service;

import com.gl.app.PackageService.payload.PackageDto;
import com.gl.app.PackageService.payload.PackageResponse;
import org.springframework.stereotype.Service;


@Service
public interface PackageService {

    public PackageDto createPackage(PackageDto packageDto);
    public PackageResponse getAllPackages();
    public PackageDto getPackageById(String packageID);
    public  PackageResponse  getPackageByCountry(String country);
}

File: Frontend/src/app/service/package.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class PackageService {
  private url = "http://localhost:8080/api/package"
  constructor(private http: HttpClient) {}
  public getAllPackages():any{
    return this.http.get(`${this.url}/getPackages`);
  }
  public getPackageById(packageID:string):any{
    return this.http.get(`${this.url}/onepackage/${packageID}`);
  };
  public getPackageByCountry(country:string):any{
    return this.http.get(`${this.url}/packages/${country}`);
  }
}
